boto3/resources/factory.py

```diff
@@ -194,6 +194,17 @@ def _load_waiters(self, attrs, model):
         for waiter in model.waiters:
             attrs[waiter.name] = self._create_waiter(waiter)
 
+    def _create_identifier(factory_self, name):
+        """
+        Creates a read-only property for identifier attributes.
+        """
+        def identifier(self):
+            return getattr(self, '_' + name, None)
```

Comment:
Should we be returning `None` if the internal attr doesn't exist?  Seems like that would only happen if we had a bug, in which case we'd want it to fail loudly.